src/Core/Resolvers/MsSqlQueryBuilder.cs

```diff
@@ -151,17 +287,50 @@ public string Build(SqlUpsertQueryStructure structure)
                 string createPredicates = JoinPredicateStrings(structure.GetDbPolicyForOperation(EntityActionOperation.Create));
 
                 // Query to insert record (if there exists none for given PK).
-                StringBuilder insertQuery = new($"INSERT INTO {tableName} ({insertColumns}) OUTPUT {outputColumns}");
-
+                StringBuilder insertQuery = new($"INSERT INTO {tableName} ({insertColumns}) ");
+
+                bool isInsertTriggerEnabled = sourceDefinition.IsInsertDMLTriggerEnabled;
+                // We can only use OUTPUT clause to return inserted data when there is no trigger enabled on the entity.
+                if (!isInsertTriggerEnabled)
+                {
+                    if (isUpdateTriggerEnabled)
+                    {
+                        // This is just an optimisation. If update trigger was not enabled, then the columnsToBeReturned would
+                        // have already been created with Inserted prefix.
+                        columnsToBeReturned = MakeOutputColumns(structure.OutputColumns, OutputQualifier.Inserted.ToString());
+                    }
+
+                    insertQuery.Append($"OUTPUT {columnsToBeReturned}");
+                }
+                // If an insert trigger is enabled but there was no update trigger enabled,
+                // we need to generated columns to be returned without 'Inserted' prefix.
+                else if (!isUpdateTriggerEnabled)
+                {
+                    // This is again just an optimisation. If update trigger was enabled, then the columnsToBeReturned would
+                    // have already been created without any prefix.
+                    columnsToBeReturned = MakeOutputColumns(structure.OutputColumns, string.Empty);
```

Comment:
Would still need to change the method signature because we need table_name (a string) in prefix here:
https://github.com/Azure/data-api-builder/blob/c3e1c32c0d12f53717a5bd9c01a4497ab52e6b38/src/Core/Resolvers/MsSqlQueryBuilder.cs#L127

Reply:
Removing the None value from the enum